compiler/expressions.go

```diff
@@ -1112,6 +1112,12 @@ func (c *funcContext) translateImplicitConversion(expr ast.Expr, desiredType typ
 	}
 
 	exprType := c.p.TypeOf(expr)
+
+	// whenever expr has an underlying *types.Interface type we need to $copyInterfaceVal
+	if _, isIface := exprType.Underlying().(*types.Interface); isIface {
```

Comment:
(Highly agreed with @neelance's remark. Adding unused code IMO is much worse than not handling some cases. Unhandled cases are obvious to spot when it happens, but unused code is not obvious and will confuse readers, adds complexity, needs to be maintained, etc.)